antenna-workflow-steps/analyzers/antenna-ort-result-analyzer/src/main/java/org/eclipse/sw360/antenna/workflow/analyzers/OrtResultAnalyzer.java

```diff
@@ -0,0 +1,247 @@
+                } else {
+                    artifact = mapArtifact(ortPackage);
+                }
+            } else {
+                artifact = mapArtifact(ortPackageIterator.next().get("package"));
+            }
+            artifacts.add(artifact);
+        }
+        return artifacts;
+    }
+
+    private Artifact mapArtifact(JsonNode ortPackage) {
+        Artifact a = new Artifact("OrtResult")
+                .addFact(mapFileName(ortPackage))
+                .addFact(new DeclaredLicenseInformation(mapDeclaredLicenses("declared_licenses", ortPackage)))
+                .addFact(new ArtifactSourceUrl(mapSourceUrl(ortPackage)));
+        mapCoordinates(ortPackage.get("id").textValue()).ifPresent(a::addFact);
+        return a;
+
+    }
+
+    private Artifact mapArtifact(JsonNode ortPackage, JsonNode scanResult) {
+        Artifact a = mapArtifact(ortPackage);
+        LicensePair artifactLicenseCopyrightInfo = mapObservedLicense(scanResult);
+        a.addFact(new ObservedLicenseInformation(artifactLicenseCopyrightInfo.getLicenseInformation()))
+                .addFact(artifactLicenseCopyrightInfo.getCopyrightStatement());
+        return a;
+    }
+
+    private Optional<JsonNode> getScanResultWithId(JsonNode ortPackage, JsonNode scanResult) {
+        JsonNode pkgId = ortPackage.get("id");
+
+        return StreamSupport.stream(scanResult.spliterator(), false)
+                .filter(currentScanResult -> currentScanResult.get("id").equals(pkgId))
+                .findFirst();
+    }
+
+    private LicenseInformation mapDeclaredLicenses(String identifier, JsonNode ortPackage) {
+        if (null != ortPackage) {
+            if (ortPackage.get(identifier).isArray()) {
+                Iterator<JsonNode> licenseIterator = ortPackage.get(identifier).elements();
+                Collection<String> licenses = new ArrayList<>();
+                while (licenseIterator.hasNext()) {
+                    licenses.add(licenseIterator.next().textValue());
+                }
+                return LicenseSupport.mapLicenses(licenses);
+            }
+        }
+        return new LicenseStatement();
+    }
+
+    private LicensePair mapObservedLicense(JsonNode scanResult) {
+        if (scanResult.get("results").isArray()) {
+            Collection<String> licenses = new ArrayList<>();
+            CopyrightStatement copyrights = new CopyrightStatement("");
+            Iterator<JsonNode> results = scanResult.get("results").elements();
+            while (results.hasNext()) {
+                Iterator<JsonNode> licenseFindings =
+                        results.next().get("summary").get("license_findings").elements();
+
+                while (licenseFindings.hasNext()) {
+                    JsonNode licenseFind = licenseFindings.next();
+                    licenses.add(licenseFind.get("license").textValue());
+                    Iterator copyrightIterator = licenseFind.get("copyrights").elements();
```

Comment:
throws NPE if `"copyrights"` is not set?

Reply:
if there are no copyrights then there will be an empty array
(like in test/resources/scan-result.yml line 124). Thus when in the next line the copyrightIterator.hasNext() is called it will just not find a next and go on. Since the output of the scan result is either an empty or a filled array, but never missing (otherwise ORT would have failed in some way) I thought it would not be necessary to do a null check here.